contracts/usul/LinearTokenVoting.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+import "./BaseTokenVoting.sol";
+import "./BaseQuorumPercent.sol";
+
+/// @title A Usul strategy that enables linear token voting
+contract LinearTokenVoting is BaseTokenVoting, BaseQuorumPercent {
+    ERC20Votes public governanceToken;
+
+    constructor(
+        address _owner,
+        ERC20Votes _governanceToken,
+        address _usulModule,
+        uint256 _votingPeriod,
+        uint256 quorumNumerator_,
+        uint256 _timelockPeriod,
+        string memory name_
+    ) {
+        bytes memory initParams = abi.encode(
+            _owner,
+            _governanceToken,
+            _usulModule,
+            _votingPeriod,
+            quorumNumerator_,
+            _timelockPeriod,
+            name_
+        );
+        setUp(initParams);
+    }
+
+    /// @notice Sets up the contract with initial parameters
+    /// @param initParams The initial setup parameters encoded as bytes
+    function setUp(bytes memory initParams) public override initializer {
```

Comment:
can this be private?

Reply:
No, this needs to be able to be called by the ModuleProxyFactory, since we are using the clone pattern here.